### DISCUSSION TITLE
🎧 Kaizen 11 - S3 R2 B2 D2 - Aug. 11, 2023

### DISCUSSION THREAD

**Discussion by gerhard** [thechangelog/changelog.com#469] [Kaizen]
[![shields.io](https://img.shields.io/badge/Last%20updated%20on%20August%2028%2C%202023-PUBLISHED-blue?style=for-the-badge)](https://changelog.com/friends/10)

[🎧 This is now available as changelog.com/friends/10](https://changelog.com/friends/10)

[![image](https://github.com/thechangelog/changelog.com/assets/3342/d7d2ed8d-8504-46f0-b39d-7b1038e88fae)](https://changelog.com/friends/10)

See all improvements in this 🐙 GitHub repository: [Kaizen 10 ... 11](https://github.com/thechangelog/changelog.com/compare/kaizen.10...kaizen.11)

---

This contains all things related to the next Kaizen episode.

Share your thoughts via comments below:
- Is there something in changelog.com that you, our dear listener, would like us to tackle?
- Is there something that **you** would like to help out with?
- Anything that we - @adamstac @jerodsanto - want to improve?

This is a rough timeline that we have in mind (red line is today):

```mermaid
gantt
    title 🎧 Kaizen 11 timeline
    dateFormat  YYYY-MM-DD
    axisFormat %b %e, %Y

    section DISCUSS
    ✍️ Add your comments below :2023-07-01, 38d

    section CONTRIBUTE
    💻 ALL CONTRIBUTIONS WELCOME 👕 :2023-07-01, 38d

    section RECORD


    🎙 Record :milestone, 2023-08-08
    ✂️ Edit :2023-08-08, 3d
    🚀 SHIP IT! :milestone, 2023-08-11

    section NEXT
    Kick-off Kaizen 12 :2023-08-11, 14d
```

---

### 🎙 Proposed topics

- Phoenix v1.7 upgrade
- PostgreSQL v15 upgrade
- Transactional Email delivery via Oban
- Migrate from AWS S3 to Cloudflare R2
  - [Extended Storage Lens](https://s3.console.aws.amazon.com/s3/lens/dashboard/default-account-dashboard?region=us-east-1)
- https://status.changelog.com
- Migration to Fly apps v2 (a.k.a. machines) is now complete
- A more resilient CI setup ✌️☝️
  - Fly.io
    - https://community.fly.io/t/fdaa4556-3-wireguard-gateway-intermittent-failure/14597
  - GitHub Actions fallback
  - K8s (Talos) 💪
- Honeycomb SLOs - all two of them!
- Fastly.com LHR POP latency increased by 50% on June 20th, 2023
  - Fixed on July 25th, 2023

#### Honeycomb SLOs

<img width="1491" alt="image" src="https://github.com/thechangelog/changelog.com/assets/3342/feda4669-b519-460d-8254-24a331539381">

- [`95%` of the time, podcast feeds should be served within `1s`](https://ui.honeycomb.io/changelog/datasets/fastly/slo/3vSeCSXS671/Podcast-feed-response-latency) (last 30 days)
- [regardless whether cached or uncached](https://ui.honeycomb.io/changelog/datasets/fastly/result/hrNYXuWsCYZ)
- [breakdown by podcast](https://ui.honeycomb.io/changelog/datasets/fastly/result/4uQGGiAUdLc)
- [`p99` - from `1s` e.g. GoTime to `11s` for Changelog & `25s` for Master](https://ui.honeycomb.io/changelog/datasets/fastly/result/8kodQ7HfARS)

<img width="1498" alt="image" src="https://github.com/thechangelog/changelog.com/assets/3342/53e970de-565c-4446-9a67-f4f76d9d8d3c">

- [`98%` of all responses are `2xx` or `3xx`](https://ui.honeycomb.io/changelog/datasets/fastly/slo/4mzdutPaJvf/2xx-or-3xx-responses) (last 30 days)
- [let's dig into this one](https://ui.honeycomb.io/changelog/datasets/fastly/result/2QcqUfxybUT)
- [what is](https://ui.honeycomb.io/changelog/datasets/fastly/result/jdwoR3ah52t)
  - `403 /images/podcasts/master-original-3922a5ce5ede81d973adc3cd618f7e64.png`
  - `403 /uploads/podcast/534/changelog-interviews-534.mp3`
  - FWIW https://github.com/thechangelog/changelog.com/issues/457#issuecomment-1503027429

**What would be a better SLO for you?**

**More than 2 SLOs?** 🤔
- [errors by library.name](https://ui.honeycomb.io/changelog/datasets/changelog_opentelemetry/result/i4hzgYzsaFM)

#### Fastly.com LHR POP network latency increase

Started June 20th. Fixed July 25th.
<img width="2807" alt="Screenshot 2023-07-26 at 13 02 39" src="https://github.com/thechangelog/changelog.com/assets/3342/1ccc32fa-c31f-4167-9548-e9b8ca920aac">
<img width="1436" alt="Screenshot 2023-07-02 at 13 18 16" src="https://github.com/thechangelog/changelog.com/assets/3342/a262de2f-5e81-47f8-ab98-f0e8cbbd49d4">

---

### What changed so far? [Kaizen.10 ... NOW](https://github.com/thechangelog/changelog.com/compare/kaizen.10...master)
- [Upgrade to Phoenix 1.7](https://github.com/thechangelog/changelog.com/commit/31831300861ab68c0fd5bca1062e2bbc651b6c32)
- [Deliver *all* mailers through Oban](https://github.com/thechangelog/changelog.com/commit/07f417c7496e847cd9e8538681a885695040fe82)
- #470
- #475
- #476 
- #477
- #471 
- #468 

### What else is coming up?
- #472
- #451
- https://middleware.io/
- 1Password Service Accounts

---

This is a follow-up to:
- https://github.com/thechangelog/changelog.com/discussions/452
- https://github.com/thechangelog/changelog.com/discussions/440
- https://github.com/thechangelog/changelog.com/discussions/433

**Comment by jerodsanto** [comment]
PR incoming that I'll need your help with @gerhard: I signed us up for Oban Web and we will need to get it all set up, which includes doing some fancy work with the deployment pipeline. Install directions are here:

https://getoban.pro/docs/web/2.9.7/installation.html

I think this could be a good Kaizen 11 topic as well since it is related to the Oban mailers task (assuming we get 'er done in time)

**Reply by gerhard** [reply]
Sounds great! Adding it to the list.

**Comment by gerhard** [comment]
FYI https://elixirforum.com/t/using-dagger-elixir-sdk-to-refactor-a-script/56748

**Comment by jerodsanto** [comment]
Kaizen 11 has shipped! 🚀 https://changelog.com/friends/10
